### PR TITLE
Merge esxcli and govc test utils to single file

### PR DESCRIPTION
**Pull request by ashahi1**
govc and esxcli are just the different interfaces that are used to communicate with ESX. Its better if we combine them into single file. Hence creating a single ESX module.

https://github.com/vmware/docker-volume-vsphere/pull/1381#discussion_r121061797

Tested locally. Failed tests are the two swarm tests which did not run as my setup didn't have swarm configured. Rest all tests passed. 

```
target e2e-test-runalways:

OK: 4 passed
--- PASS: Test (98.12s)
PASS

----------------

target e2e-test-runonce:

OOPS: 15 passed, 2 FAILED, 1 MISSED
--- FAIL: Test (242.50s)
FAIL
exit status 1
FAIL	github.com/vmware/docker-volume-vsphere/tests/e2e	242.502s
make: *** [e2e-test-runonce] Error 1
make: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
make[1]: *** [e2e-test] Error 2
make: *** [e2e-test] Error 2

```

Failed tests

### CODE REVIEW

**Comment by shuklanirdesh82** [comment]
> Tested locally.

Good but not informative, it is good to add some short of test run output. How was the change tested?

**Comment by ashahi1** [comment]
> https://github.com/vmware/docker-volume-vsphere/pull/1426#issuecomment-308613173

@shuklanirdesh82 I have updated the test output in the summary.

File: tests/e2e/volumecreate_test.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"strings"
 	"sync"
 
+	dockerclicon "github.com/vmware/docker-volume-vsphere/tests/constants/dockercli"
```
Comment (govint) [inline tests/e2e/volumecreate_test.go]: dockercliconst?

File: tests/utils/esxcli/cmd.go [hunk 2]
```diff
@@ -47,3 +56,55 @@ func KillVM(esxHostIP, worldID string) bool {
 	}
 	return true
 }
+
+// RetrieveVMNameFromIP util retrieves VM  name from passed VM IP
+//esxcli vm.info -vm.ip=10.20.104.62 -json | jq -r .VirtualMachines[].Name
```
Comment (govint) [inline tests/utils/esxcli/cmd.go]: esxcli vm.info -vm.ip=<ip> -json | jq -r .VirtualMachines[].Name

File: tests/utils/esxcli/cmd.go [hunk 3]
```diff
@@ -47,3 +56,55 @@ func KillVM(esxHostIP, worldID string) bool {
...
+func RetrieveVMNameFromIP(ip string) string {
+	log.Printf("Finding VM name from IP Address [%s]\n", ip)
```
Comment (govint) [inline tests/utils/esxcli/cmd.go]: For all these utility functions, lets not put a log for each function. Once we have debug logs for tests then we can add these. For now please leave the printf() calls but leave them commented out. If some test is misbehaving then the developer can enable that log and test. Lets not have these logs turned on.

File: tests/utils/esxcli/cmd.go [hunk 4]
```diff
@@ -47,3 +56,55 @@ func KillVM(esxHostIP, worldID string) bool {
...
+	cmd := esxcli.VMInfoByIP + ip + esxcli.JSONTypeOutput + "| " + esxcli.JSONParser + esxcli.VMName
```
Comment (govint) [inline tests/utils/esxcli/cmd.go]: Can "esxcli.JSONTypeOutput + vmName + " | " + esxcli.JSONParser" be a constant in this file and use it here and in other functions.

File: tests/utils/esxcli/cmd.go [hunk 5]
```diff
@@ -47,3 +56,55 @@ func KillVM(esxHostIP, worldID string) bool {
...
+func WaitForExpectedState(fn getVMPowerStatus, vmName, expectedState string) bool {
+	log.Printf("Confirming [%s] power status for vm [%s]\n", expectedState, vmName)
+	for attempt := 0; attempt < maxAttempt; attempt++ {
+		misc.SleepForSec(waitTime)
+		status := fn(vmName)
+		if status == expectedState {
+			return true
+		}
+	}
+	log.Printf("Timed out to poll status\n")
```
Comment (govint) [inline tests/utils/esxcli/cmd.go]: Lets keep this printf as its logging an error.

File: tests/constants/esxcli/cmd.go [hunk 6]
```diff
@@ -17,7 +17,8 @@
 package esxcli
```
Comment (shuklanirdesh82) [inline tests/constants/esxcli/cmd.go]: General comment: As we have decided to merge ... I would prefer something like `vsphereutil` certainly not `esxcli`

File: tests/utils/esxcli/cmd.go [hunk 7]
```diff
@@ -18,15 +18,25 @@ package esxcli
...
+	"strings"
 
 	"github.com/vmware/docker-volume-vsphere/tests/constants/esxcli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
 )
 
+type getVMPowerStatus func(string) string
+
+const (
```
Comment (shaominchen) [inline tests/utils/esxcli/cmd.go]: As we have discussed before, let's avoid defining global constants in local files. If these constants will be shared, we should define them in the global constants package. If it's just for local use in the same file, we can define them as local variables. Global variables will pollute the package namespace even if they are unexported.

**Comment by ashahi1** [comment]
I created a single file for esx and govc methods as it was mentioned during my separate review https://github.com/vmware/docker-volume-vsphere/pull/1381#discussion_r121061797

**Comment by ashahi1** [comment]
https://github.com/vmware/docker-volume-vsphere/pull/1426#pullrequestreview-44384944 
@govint Shall we have single file for govc and esxcli functions or have a single package and keep both files under that package? Please suggest.

/cc @shuklanirdesh82

**Comment by govint** [comment]
A single (with clear separation of the two esxcli and govc) or multiple files are both ok. If we don't see the govc functions growing in number (if we can see that, that is) then may be have a single file else go for two.

**Comment by ashahi1** [comment]
Making the change to have single package with two files.

**Comment by shuklanirdesh82** [comment]
CI has passed and two approval. Merging this PR.

**Pull request merged**